File: src/app/models/project.ts
import { AgentRule } from "./agent.rule";
import { Task } from "./task";

export interface Project {
    id: string;
    name: string;
    path: string;
    rules: AgentRule[];
    tasks: Task[];
    createdAt: number;
    updatedAt: number;
}

export const EMPTY_PROJECT: Project = {
    id: '',
    name: '',
    path: '',
    rules: [],
    tasks: [],
    createdAt: 0,
    updatedAt: 0,
};

File: src-tauri/src/models/project.rs
use serde::{Deserialize, Serialize};

use super::agent_rule::AgentRule;
use super::task::Task;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Project {
    pub id: String,
    pub name: String,
    pub path: String,
    pub rules: Vec<AgentRule>,
    pub tasks: Vec<Task>,
    pub created_at: i64,
    pub updated_at: i64,
}
